x-pack/plugins/security_solution/public/common/components/event_details/related_cases.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo, useState, useEffect } from 'react';
+import { EuiFlexItem, EuiLoadingContent } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useKibana } from '../../../common/lib/kibana';
+import { TimelineEventsDetailsItem } from '../../../../common/search_strategy';
+import { getFieldValue } from '../../../detections/components/host_isolation/helpers';
+import { CaseDetailsLink } from '../links';
+import { APP_ID } from '../../../../common/constants';
+
+interface Props {
+  data: TimelineEventsDetailsItem[];
+  isAlert: boolean;
+}
+
+type RelatedCaseList = Array<{ id: string; title: string }>;
+
+export const RelatedCases: React.FC<Props> = React.memo(({ data, isAlert }) => {
+  const {
+    services: { cases },
+  } = useKibana();
+
+  const alertId = useMemo(() => getFieldValue({ category: '_id', field: '_id' }, data), [data]);
+
+  const [relatedCases, setRelatedCases] = useState<RelatedCaseList>([]);
+  const [areCasesLoading, setAreCasesLoading] = useState(true);
+
+  const getRelatedCases = useCallback(async () => {
+    let relatedCaseList: RelatedCaseList = [];
+    try {
+      relatedCaseList =
+        (await cases.api.getRelatedCases(alertId, {
+          owner: APP_ID,
+        })) ?? [];
+    } catch {
+      // TODO: Show an error toaster here?
```

Comment:
When the code is modified (locally) to simulate a failure by always throwing an error when data is fetched, the loading indicator stops, and `This alert was found in 0 cases.` is displayed:

![simulated_failure](https://user-images.githubusercontent.com/4459398/159045387-4d72b402-c693-4b97-b056-aa9ebd98d457.png)

If the API is call is not expected to fail often, it seems reasonable to display an error toaster.

If the API call is expected to fail often, such that an error toaster might be annoying to the user, consider displaying an alternative to `This alert was found in 0 cases.` such that:

- It conveys the fact we couldn't make this determination
- It has an action that enables the user to view the details explaining why

Reply:
Agreed. @monina-n would you prefer us to show a toaster if the query for the related cases fails?

Reply:
Spoke with @monina-n offline about this and we decided to not show it if there is an error and provide a warning toast. Also added the check for cases read permissions as that was missing previously. Thanks again!